frmts/saga/sagadataset.cpp

```diff
@@ -773,25 +797,15 @@ CPLErr SAGADataset::SetGeoTransform( double *padfGeoTransform )
     const double dfMinY =
         padfGeoTransform[5] * (nRasterYSize - 0.5) + padfGeoTransform[3];
 
-    const CPLString osPath = CPLGetPath( GetDescription() );
-    const CPLString osName = CPLGetBasename( GetDescription() );
-    const CPLString osHDRFilename = CPLFormCIFilename( osPath, osName, ".sgrd" );
 
-    CPLErr eErr = WriteHeader( osHDRFilename, poGRB->GetRasterDataType(),
-                               poGRB->nRasterXSize, poGRB->nRasterYSize,
-                               dfMinX, dfMinY, padfGeoTransform[1],
-                               poGRB->m_NoData, 1.0, false );
-
-    if( eErr == CE_None )
-    {
-        poGRB->m_Xmin = dfMinX;
-        poGRB->m_Ymin = dfMinY;
-        poGRB->m_Cellsize = padfGeoTransform[1];
-        poGRB->m_Cols = nRasterXSize;
-        poGRB->m_Rows = nRasterYSize;
-    }
-
-    return eErr;
+    poGRB->m_Xmin = dfMinX;
+    poGRB->m_Ymin = dfMinY;
+    poGRB->m_Cellsize = padfGeoTransform[1];
+    poGRB->m_Cols = nRasterXSize;
```

Comment:
I know this was there before but setting m_nCols and m_nRows in this method is weird. My understanding of the Create() and CreateCopy() implementation is that they should already be set (when Create() calls Open() at the end, which will read the serialized initial header)

Reply:
I had a look, and actually m_nCols and m_nRows are not used at all. I will remove them.